packages/core/src/storage/namespace.ts

```diff
@@ -140,6 +160,52 @@ function addContractNamespacesWithSrc(
   }
 }
 
+/**
+ * Checks if a custom layout's base slot clashes with a namespace's base slot, and throws an error if it does.
+ *
+ * @param contractDef The contract definition to check.
+ * @param storageLocation The namespace storage location string, e.g. `erc7201:<NAMESPACE_ID>`.
+ * @param contractName The name of the contract.
+ * @param src The original source location of the struct with the namespace.
+ * @throws CustomLayoutClashWithNamespaceError if the custom layout clashes with the namespace.
+ */
+function checkCustomLayoutClashWithNamespace(
+  contractDef: ContractDefinition,
+  storageLocation: string,
+  contractName: string,
+  src: string,
+) {
+  if (contractDef.storageLayout !== undefined && storageLocation.startsWith(ERC7201_FORMULA_PREFIX)) {
+    const namespaceId = storageLocation.substring(ERC7201_FORMULA_PREFIX.length);
+    const storageLocationHash = calculateERC7201StorageLocation(namespaceId);
```

Comment:
nit: this is not a hash (i.e. is `hash() & `~bytes32(uint256(0xff)`), so I would just call it `calculatedStorageLocation` or similar